tests/unit/test_business_logic.py

```diff
@@ -384,6 +384,23 @@ def test_set_status(current, future, valid_author, valid_checker, bll):
             bll.set_status(release_request1, future, user=author)
 
     if valid_checker:
+        if current == RequestStatus.SUBMITTED:
```

Comment:
The number of `if` branches in this test make me feel like it should be split up into 2 or 3 separate ones now

Reply:
(not necessarily for this PR)

Reply:
yes, I think this could definitely use a refactor, I'll do in a follow-up